addons/website_slides/models/slide_slide.py

```diff
@@ -700,165 +816,231 @@ def _compute_quiz_info(self, target_partner, quiz_done=False):
-        except json.decoder.JSONDecodeError:
-            error = str(error)
+            thumbnail_url = snippet['thumbnails']['high']['url']
+            if fetch_image:
+                slide_metadata['image_1920'] = base64.b64encode(
+                    requests.get(thumbnail_url, timeout=3).content
+                )
+            else:
+                slide_metadata['image_url'] = thumbnail_url
 
-        if error == 'keyInvalid':
-            return _('Your Google API key is invalid, please update it in your settings.\nSettings > Website > Features > API Key')
+        return slide_metadata
 
-        return _('Could not fetch data from url. Document or access right not available:\n%s', error)
+    def _fetch_google_drive_metadata(self, fetch_image, return_when_not_found=False):
+        """ Fetches document / video metadata from the Google Drive API.
+
+        Returns a dict containing metadata with the following keys (matching slide.slide fields):
+        - 'name' matching the external file title
+        - 'image_1920' binary data of the file thumbnail
+          OR 'image_url' containing an external link to the thumbnail when 'fetch_image' param is False
+        - 'completion_time' which is computed for 2 types of files:
+          - pdf files where we download the content and then use slide.slide#_get_completion_time_pdf()
+          - videos where we use the 'videoMediaMetadata' to extract the 'durationMillis'
+
+        :param fetch_image: if False, will return 'image_url' instead of binary data
+          Typically used when displaying a slide preview to the end user.
+        :param return_when_not_found: if True, will return the error in case file cannot be found. """
 
-    @api.model
-    def _parse_google_document(self, document_id, only_preview_fields):
-        def get_slide_category(vals):
-            # TDE FIXME: WTF ??
-            slide_category = 'presentation'
-            if vals.get('image_1920'):
-                image = Image.open(io.BytesIO(base64.b64decode(vals['image_1920'])))
-                width, height = image.size
-                if height > width:
-                    return 'document'
-            return slide_category
-
-        # Google drive doesn't use a simple API key to access the data, but requires an access
-        # token. However, this token is generated in module google_drive, which is not in the
-        # dependencies of website_slides. We still keep the 'key' parameter just in case, but that
-        # is probably useless.
         params = {}
         params['projection'] = 'BASIC'
         if 'google.drive.config' in self.env:
-            access_token = self.env['google.drive.config'].get_access_token()
+            access_token = False
+            try:
+                access_token = self.env['google.drive.config'].get_access_token()
+            except RedirectWarning as e:
+                pass  # ignore and use the 'key' fallback
+            except UserError as e:
```

Comment:
I think except can be chained (except A, B )

Reply:
You have to "tuplelize" them to make it work

`except (A,B)`

Learned something today yay.